bin/backfill-location-hierarchy

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+"""
+Creates an ncov-ingest style location hierarchy TSV using data from
+nextstrain/fauna's ./source-data/geo_synonyms.tsv and
+./source-data/geo_locations.tsv files as well as the currently up-to-date
+(ncov-ingest) GISAID metadata TSV on S3.
+
+This script is meant to be run only once to create a local
+./source-data/location_hierarchy.tsv as a starting point for a source of truth
+for location hierarchies in all future metadata.
+"""
+import sys
+import argparse
+import pandas as pd
+from pathlib import Path
+sys.path.insert(0, str(Path(__file__).parent.parent / "lib"))
+from utils.hierarchy_dataframe import hierarchy_dataframe
+
+
+FAUNA_PREFIX = 'https://raw.githubusercontent.com/nextstrain/fauna/master/source-data/'
+GEO_SYNONYMS = FAUNA_PREFIX + 'geo_synonyms.tsv'
+GEO_REGIONS = FAUNA_PREFIX + 'geo_regions.tsv'
+LOCATION_HIERARCHY_COLUMNS = ['region', 'country', 'division']
+
+
+def prepare_fauna_data() -> pd.DataFrame:
+    """
+    Performs some standardizing of Fauna's "region" data, then merges fauna's
+    geo_synonyms.tsv and geo_regions.tsv into one file.
+    """
+    synonyms = pd.read_csv(GEO_SYNONYMS, sep='\t', comment='#')
+    regions = pd.read_csv(GEO_REGIONS, sep='\t')
+
+    synonyms['country_lower'] = synonyms['country'].str.lower()
+
+    regions = regions.rename(columns={'country': 'country_lower'})
+    regions['region'] = regions['region'].str.replace('_', ' ').str.title()
+
+    return pd.merge(regions, synonyms, on='country_lower', validate='1:m') \
+        [LOCATION_HIERARCHY_COLUMNS] \
+        .dropna()
+
+
+if __name__ == '__main__':
+    base = Path(__file__).resolve().parent.parent
+
+    parser = argparse.ArgumentParser(
+        description="Create a location hierarchy TSV file from fauna's "
+            "./source-data/geo_*.tsv files and ncov-ingest's current GISAID "
+            "metadata.tsv on S3",
+        formatter_class=argparse.RawTextHelpFormatter
+    )
+    parser.add_argument("metadata",
+        default="s3://nextstrain-ncov-private/metadata.tsv.gz",
+        nargs="?",
+        help="A GISAID metadata TSV file. \nDefaults to "
+            "s3://nextstrain-ncov-private/metadata.tsv.gz.")
+    parser.add_argument("--output",
+        default=base / "source-data/location_hierarchy.tsv",
+        nargs="?",
+        help="Output location for the new location hierarchy TSV file.\n"
+            "Defaults to ../source-data/location_hierarchy.tsv.")
```

Comment:
I'm not sure what's an appropriate default value to put in the help string here. Should it be relative to this file? The top of the repo? Should I use string substitution and replace it with a path on the user's computer?